src/utils/num_utils.ts

```diff
@@ -78,3 +85,118 @@ export function formatNumber(value: number, sigFigs = DEFAULT_SIG_FIGS, sciSigFi
     return trimmed.endsWith(".") ? trimmed.slice(0, -1) : trimmed;
   }
 }
+
+export function timeToMilliseconds(time: number, unit: TimeUnit): number {
+  if (unit == TimeUnit.MILLISECOND) {
+    return time;
+  } else if (unit == TimeUnit.SECOND) {
+    return time * SECONDS_IN_MS;
+  } else if (unit == TimeUnit.MINUTE) {
+    return time * MINUTES_IN_MS;
+  } else if (unit == TimeUnit.HOUR) {
+    return time * HOURS_IN_MS;
+  } else if (unit == TimeUnit.DAY) {
+    return time * DAYS_IN_MS;
+  } else {
+    throw new Error("Unrecognized time unit");
+  }
+}
```

Comment:
Maybe use an object to handle the time unit here to avoid multiple `else if`? 
something like:
```
const timeUnitToMilliseconds = {
  [TimeUnit.MILLISECOND]: 1,
  [TimeUnit.SECOND]: SECONDS_IN_MS,
  [TimeUnit.MINUTE]: MINUTES_IN_MS,
  [TimeUnit.HOUR]: HOURS_IN_MS,
  [TimeUnit.DAY]: DAYS_IN_MS,
};

export function timeToMilliseconds(time: number, unit: TimeUnit): number {
  const timeUnitFactor = timeUnitToMilliseconds[unit];
  if (timeUnitFactor === undefined) {
    throw new Error("Unrecognized time unit");
  }
  return time * timeUnitFactor;
}
```

Reply:
Makes sense, thank you!